### PR TITLE
feature_LWDEV-7723-show-indicative-total: added indicative total pric…

### PR DESCRIPTION
**Pull request by Georg3MK**
…e for Market order

### CODE REVIEW

File: src/components/OrderBook/LevelList.tsx [hunk 1]
```diff
@@ -141,16 +142,23 @@ class LevelList extends React.Component<LevelListProps> {
     );
 
     levels.forEach((l, i: number) => {
```
Comment (GeorgePuisha) [inline src/components/OrderBook/LevelList.tsx]: I think that this code should be slightly refactored. `level`, `index` are better.

File: src/models/mappers/orderMapper.ts [hunk 2]
```diff
@@ -11,3 +11,55 @@ export const toOrder = (dto: any, side: Side = Side.Buy) =>
...
+
+const getValueForAvailableVolume = (order: Order) => order.price * order.volume;
+
+const getAvailableVolume = (amount: number, order: Order) =>
+  amount * order.volume / order.price;
+
+export const mapToEffectivePrice = (volume: number, orders: Order[]) => {
+  let price: number = 0;
+
+  const volumeSum = orders.reduce((sum, order) => {
```
Comment (GeorgePuisha) [inline src/models/mappers/orderMapper.ts]: Can be moved upper, too long to be an arrow function

File: src/components/Order/Order.tsx [hunk 3]
```diff
@@ -73,6 +74,14 @@ interface OrderProps {
   resetOrder: () => void;
   isDisclaimerShown: boolean;
   disclaimedAssets: string[];
+  setMarketTotal: (
+    operationVolume?: any,
+    operationType?: string,
+    debounce?: boolean
+  ) => any;
```
Comment (PaulRumyantsevItr) [inline src/components/Order/Order.tsx]: any? You sure?

File: src/components/Order/OrderMarket.tsx [hunk 4]
```diff
@@ -23,7 +27,11 @@ interface OrderMarketState {
 }
 
 export interface OrderMarketProps extends OrderBasicFormProps {
+  amount: string;
+  setMarketTotal: (volume?: any, action?: string, debounce?: boolean) => any;
```
Comment (PaulRumyantsevItr) [inline src/components/Order/OrderMarket.tsx]: The same

File: src/components/Order/OrderMarket.tsx [hunk 5]
```diff
@@ -117,6 +132,19 @@ class OrderMarket extends React.Component<
...
           ))}
         </Flex>
+        <Total>
+          <OrderTitle>
+            Total
+            <TotalHint title={'Your order may execute at a different price'}>
+              Indicative price *
+            </TotalHint>
+          </OrderTitle>
+          {!isNotEnoughLiquidity && (
```
Comment (PaulRumyantsevItr) [inline src/components/Order/OrderMarket.tsx]: {isEnoughLiquidity && ...

File: src/stores/uiOrderStore.ts [hunk 6]
```diff
@@ -212,6 +240,55 @@ class UiOrderStore extends BaseStore {
...
+    operationVolume?: any,
+    operationType?: Side,
+    debounce?: boolean
+  ) => {
+    if (operationVolume === '') {
+      return this.resetMarketTotal();
+    }
+
+    const areNewValues = operationVolume && operationType;
+    const areNotNewValues = !operationVolume && !operationType;
```
Comment (PaulRumyantsevItr) [inline src/stores/uiOrderStore.ts]: areOldValues? :)

**Pull request merged**